### ISSUE TITLE
Filtered Format: Option to make filters interactive

### ISSUE THREAD

**Issue by krabina** [SemanticMediaWiki/SemanticResultFormats#487]
### Issue
Currently the filtered format does not have the capabilty changing the filters in accordance to a selection.

### Desired behaviour
Feature 1 is a switch that makes filters interactive. This means that after selecting one filter, all the other filters should only be displayed if the results still have values where the filters have effect. Consider as example results for pages of organizations with filters country, size, and services. If I click on a filter "Austria" the result set would only show me the organizations from Austria, but still ALL services-filters are shown, even if this means that when I click on a single service, I will get zero results in the result table. This is intuitive behaviour and correct. But looking at the Semantic Drilldown extension things are different there: if you would select "Austria" as one filter, the other filter options would only show these elements where there are still results (even indicting the number of results next to the filter.

Feature 2 (related to the above mentionied) is to be able to show in brackets next to the filter the number of result values available for the filter.

Feature 3 is to make filters dependent on one another. E. g a filter "city" that is dependent on the filter "country". So only if the filter "country" is set, the filter "city" will be shown with its values (e. g. 10 cities in Austria that are in the result set).  This beheviour also can be seen in the Semantic Drilldown extension and is currently missing in the filtered format.

**Comment by s7eph4n** [comment]
Does this only concern the value filter? If not, could you give an example how other filters would work?

**Comment by krabina** [comment]
No, this concerns every filter, I guess. If you filter for a number, let's say a year and the results are drilled down to all the entries that have year=2018 thatn with the described option the filter values than can be selected are reduced to those where the filtered results still have something to filter.

E. g. if you have keywords for the entries "starting", "in progress", "finished". All there three values will be shown after the filter "year=2018" was selected even if non of those entries have a keyword "starting". 

If you click on "starting" zero results will be displayed. This is fine and intuitive, but what Semantic Drillown does is also great: only showing those filter options that available in the selected results.
